pkg/mdformatter/mdgen/mdgen.go

```diff
@@ -100,6 +111,53 @@ func (t *genCodeBlockTransformer) TransformCodeBlock(ctx mdformatter.SourceConte
 			return nil, errors.Errorf("expected language a first element of info string got %q; Got info string %q", lang, string(infoString))
 		}
 	}
+
+	// Code fence with file.
+	fileToCopy, fileOk := infoStringAttr[infoStringKeyFile]
+	linesToCopy, lineOk := infoStringAttr[infoStringKeyLines]
+	if fileOk {
+		file, err := os.Open(fileToCopy)
+		if err != nil {
+			return nil, errors.Errorf("file could not be opened, ensure file path: %q is correct. Got info string %q", infoStringKeyFile, string(infoString))
+		}
+		defer file.Close()
+
+		scanner := bufio.NewScanner(file)
+		scanner.Split(bufio.ScanLines)
+		var text []byte
+
+		switch lineOk {
+		case true:
+			val := strings.Split(linesToCopy, "-")
```

Comment:
I would propose the same semantics as Go slice, WDYT?
```suggestion
			val := strings.Split(linesToCopy, ":")
```

Reply:
Sure! I think this would be better!